modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/index.js

```diff
@@ -0,0 +1,239 @@
+				sessionId
+			);
+			if (items) {
+				setActivationKeys(items);
+				setStatusBar({
+					activeTotalCount: items.filter((activationKey) =>
+						ACTIVATION_KEYS_LICENSE_FILTER_TYPES.active(
+							activationKey
+						)
+					).length,
+					allTotalCount: items.length,
+					expiredTotalCount: items.filter((activationKey) =>
+						ACTIVATION_KEYS_LICENSE_FILTER_TYPES.expired(
+							activationKey
+						)
+					).length,
+					notActiveTotalCount: items.filter((activationKey) =>
+						ACTIVATION_KEYS_LICENSE_FILTER_TYPES.notActivated(
+							activationKey
+						)
+					).length,
+				});
+			}
+
+			setIsLoadingActivationKeys(false);
+		};
+		fetchActivationKeysData();
+	}, [licenseKeyDownloadURL, project, sessionId]);
+
+	useEffect(() => {
+		if (activationKeys.length) {
+			const activationKeysFilterData = activationKeys.filter(
+				(activationKey) =>
+					ACTIVATION_KEYS_LICENSE_FILTER_TYPES[filterStatusBar]
+						? ACTIVATION_KEYS_LICENSE_FILTER_TYPES[filterStatusBar](
+								activationKey
+						  )
+						: Boolean
+			);
+			setTotalCount(activationKeysFilterData.length || 5);
+			const activationKeysFilterByPage = activationKeysFilterData.slice(
+				itemsPerPage * activePage - itemsPerPage,
+				itemsPerPage * activePage
+			);
+			setActivationKeysFiltered(
+				activationKeysFilterByPage.length
+					? activationKeysFilterByPage
+					: activationKeysFilterData
+			);
+		}
+	}, [activationKeys, activePage, filterStatusBar, itemsPerPage]);
+
+	return (
+		<div>
+			<div className="align-center cp-dxp-activation-key-container d-flex justify-content-between">
+				<h3 className="m-0">Activation Keys</h3>
+
+				<RoundedGroupButtons
+					groupButtons={[
+						getGroupButtons(
+							ACTIVATION_STATUS.all,
+							statusBar?.allTotalCount
+						),
+						getGroupButtons(
+							ACTIVATION_STATUS.active,
+							statusBar?.activeTotalCount
+						),
+						getGroupButtons(
+							ACTIVATION_STATUS.notActivated,
+							statusBar?.notActiveTotalCount
+						),
+						getGroupButtons(
+							ACTIVATION_STATUS.expired,
+							statusBar?.expiredTotalCount
+						),
+					]}
+					handleOnChange={(value) => {
+						setFilterStatusBar(value);
+					}}
+				/>
+			</div>
+
+			<ClayTooltipProvider
+				contentRenderer={(props) => {
+					return getTooltipTitles(props);
+				}}
+				delay={100}
+			>
+				<Table
+					className="border-0 cp-dxp-activation-key-table mt-5"
+					columns={COLUMNS}
+					hasCheckbox
+					hasPagination
+					isLoading={isLoadingActivationKeys}
+					paginationConfig={{
+						activePage,
+						itemsPerPage,
+						labels: {
+							paginationResults: 'Showing {0} to {1} of {2}',
+							perPageItems: 'Show {0} Items',
+							selectPerPageItems: '{0} Items',
+						},
+						listItemsPerPage: [
+							{label: 5},
+							{label: 10},
+							{label: 20},
+							{label: 50},
+						],
+						setActivePage,
+						setItemsPerPage,
+						showDeltasDropDown: true,
+						totalCount,
+					}}
```

Comment:
Is it possible move the logic for a specific variable?
```suggestion
					paginationConfig={paginationConfig}
```